.circleci/config.yml

```diff
@@ -1,28 +1,17 @@
 version: 2
 jobs:
   build:
-    environment:
     docker:
-      - image: python:3.6-alpine
-        environment:
-        - DATABASE_URL: postgresql://root:@localhost/circle_test?sslmode=disable
-        - DEBUG: False
-        - DJANGO_SETTINGS_MODULE: backend.settings.local
-        - OIDC_OP_DOMAIN: https://auth.mozilla.auth0.com
-        - OIDC_RP_CLIENT_ID: test
-        - OIDC_RP_CLIENT_SECRET: test
-      - image: circleci/postgres:9.5
-        environment:
-        - POSTGRES_USER: root
-        - POSTGRES_DB: circle_test
+      - image: docker/compose:1.21.2
     steps:
+      - setup_remote_docker
+
       - checkout
-      - run:
-          name: Install system dependencies
-          command: apk --no-cache add build-base bash curl openssl-dev libffi-dev py-cffi postgresql-client postgresql-dev
-      - run:
-          name: Install Python libraries
-          command: pip install -U pip && pip install -r requirements.txt
-      - run: flake8 backend
-      - run: coverage run ./manage.py test
-      - run: coverage xml && curl -s https://codecov.io/bash | bash -
+
+      - run: cp backend.env.dist backend.env
+
+      - run: docker-compose run app flake8 backend
+
+      - run: docker-compose run app python manage.py test
+#      - run: coverage run ./manage.py test
+#      - run: coverage xml && curl -s https://codecov.io/bash | bash -
```

Comment:
To be able to report test coverage to Codecov you'll have to pass in the CI environment variables into the like in the [ATMO test script](https://github.com/mozilla/telemetry-analysis-service/blob/master/bin/test#L10-L13) and then submit it using the Codecov bash integration inside the container like in the [ATMO run script](https://github.com/mozilla/telemetry-analysis-service/blob/master/bin/run#L59-L62).

The reasoning is that Codecov detects a bunch of variables outside the container when CircleCI runs, but needs the info *inside* to submit it.

Since I'm not sure if consecutive runs to `docker-compose run app` will persist on CircleCI you may want to move all the steps (1. run tests, 2. generate coverage data, 3. submit coverage data to Codecov) in a script that you call with one docker-compose call.

Reply:
I'm adding a TODO for this and will need to circle back around (pun intended). Otherwise I won't finish my quarterly goal.